contracts/FindMarket.cdc

```diff
@@ -1234,26 +1336,19 @@ pub contract FindMarket {
 
 		let buyer=saleItem.getBuyer()
 		let seller=saleItem.getSeller()
-		let oldProfile= getAccount(seller).getCapability<&{Profile.Public}>(Profile.publicPath).borrow()!
 		let soldFor=vault.balance
 		let ftType=vault.getType()
 
 		/* Residual Royalty */
 		var payInFUT = false
 		var payInDUC = false
 		let ftInfo = FTRegistry.getFTInfoByTypeIdentifier(ftType.identifier)! // If this panic, there is sth wrong in FT set up
-		var residualAddress = FindMarket.residualAddress
-		if FindUtils.contains(ftType.identifier, element: "FlowUtilityToken.Vault") {
-			residualAddress = dapperMerchAddress
-			payInFUT = true
-		} else if FindUtils.contains(ftType.identifier, element: "DapperUtilityCoin.Vault") {
-			residualAddress = dapperMerchAddress
-			payInDUC = true
-		}
-		let residualVault = getAccount(residualAddress).getCapability<&{FungibleToken.Receiver}>(ftInfo.receiverPath)
+		let oldProfileCap= getAccount(seller).getCapability<&{FungibleToken.Receiver}>(Profile.publicReceiverPath)
+		let oldProfile = self.getPaymentWallet(oldProfileCap, ftInfo, panicOnFailCheck: true)
```

Comment:
the seller profile here will also go through the getPaymentWallet check. 
Currently we by default get the profile capability for deposit. 
But if they do not have some of the wallet path set, getPaymentWallet will help to get that from standard path / switchboard if possible.